PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,478 @@
+      if (pt_gam < 20.)
+        continue;
+      if (fabs(eta_gam) < 2.)
+        continue;
+      if (fabs(eta_gam) > 3.)
+        continue;
+      double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+      nonPrefiringProba[fluct] *= (1. - prefiringprob_gam);
+      nonPrefiringProbaPhoton[fluct] *= (1. - prefiringprob_gam);
+    }
+
+    //Now applying the prefiring maps to jets in the affected regions.
+    for (const auto& jet : theJets) {
+      double pt_jet = jet.pt();
+      double eta_jet = jet.eta();
+      double phi_jet = jet.phi();
+      if (pt_jet < 20.)
+        continue;
+      if (fabs(eta_jet) < 2.)
+        continue;
+      if (fabs(eta_jet) > 3.)
+        continue;
+      if (jetMaxMuonFraction_ > 0 && jet.muonEnergyFraction() > jetMaxMuonFraction_)
+        continue;
+      //Loop over photons to remove overlap
+      double nonprefiringprobfromoverlappingphotons = 1.;
+      for (const auto& photon : thePhotons) {
+        double pt_gam = photon.pt();
+        double eta_gam = photon.eta();
+        double phi_gam = photon.phi();
+        if (pt_gam < 20.)
+          continue;
+        if (fabs(eta_gam) < 2.)
+          continue;
+        if (fabs(eta_gam) > 3.)
+          continue;
+        double dR = reco::deltaR(eta_jet, phi_jet, eta_gam, phi_gam);
+        if (dR > 0.4)
+          continue;
+        double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+        nonprefiringprobfromoverlappingphotons *= (1. - prefiringprob_gam);
+      }
+      //useEMpt =true if one wants to use maps parametrized vs Jet EM pt instead of pt.
+      if (useEMpt_)
+        pt_jet *= (jet.neutralEmEnergyFraction() + jet.chargedEmEnergyFraction());
+      double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRate(eta_jet, pt_jet, h_prefmap_jet, fluct);
+
+      if (nonprefiringprobfromoverlappingphotons == 1.) {
+        nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet;
+        nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet;
+      }
+      //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+      else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet) {
+        if (nonprefiringprobfromoverlappingphotons != 0.) {
+          nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+          nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+        } else {
+          nonPrefiringProba[fluct] = 0.;
+          nonPrefiringProbaJet[fluct] = 0.;
+        }
+      }
+      //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
+    }
+    for (const auto& muon : theMuons) {
+      double pt = muon.pt();
+      double phi = muon.eta();
```

Comment:
there is bug here 
double phi = muon.phi();

Reply:
Fixed.